### PR TITLE
Swift: add DotSyntaxCallExpr tests

### PR DESCRIPTION
**Pull request by redsun82**

### CODE REVIEW

File: swift/ql/test/extractor-tests/generated/expr/DotSyntaxCallExpr/DotSyntaxCallExpr_getArgument.expected [hunk 1]
```diff
@@ -0,0 +1,2 @@
+| dot_syntax_call.swift:6:1:6:3 | call to foo(_:_:) | 0 | : X.Type |
+| dot_syntax_call.swift:7:1:7:3 | call to bar() | 0 | : X.Type |
```
Comment (geoffw0) [inline swift/ql/test/extractor-tests/generated/expr/DotSyntaxCallExpr/DotSyntaxCallExpr_getArgument.expected:2]: Why do we only see argument `0` for the `DotSyntaxCallExpr` on line 6?

Comment (redsun82) [reply swift/ql/test/extractor-tests/generated/expr/DotSyntaxCallExpr/DotSyntaxCallExpr_getArgument.expected:2]: just like with instance method calls, it would seem there is some currying going on.

So `X.foo(1, 2)` gets parsed by the swift frontend library as 

```
CallExpr(
  DotSyntaxCallExpr(
    DeclRefExpr(foo), 
    [TypeExpr(X)]
  ),
  [1, 2]
)
``` 

There is an ongoing discussion to make this more user friendly using an IPA layer on top of the generated DB types ([internal link here](https://github.com/github/codeql-c-team/issues/1019))

Comment (geoffw0) [reply swift/ql/test/extractor-tests/generated/expr/DotSyntaxCallExpr/DotSyntaxCallExpr_getArgument.expected:2]: Makes sense.  It would be good to make things more user friendly, though I worry there could be edge cases that are difficult to model another way.

**Review by geoffw0 — APPROVED** [review]
LGTM :+1:
